public/app/features/alerting/unified/components/rules/RuleStats.tsx

```diff
@@ -1,140 +1,138 @@
 import pluralize from 'pluralize';
-import React, { Fragment, useState } from 'react';
-import { useDebounce } from 'react-use';
+import React, { Fragment } from 'react';
 
 import { Stack } from '@grafana/experimental';
 import { Badge } from '@grafana/ui';
-import { CombinedRule, CombinedRuleGroup, CombinedRuleNamespace } from 'app/types/unified-alerting';
+import {
+  AlertGroupTotals,
+  AlertInstanceTotalState,
+  CombinedRuleGroup,
+  CombinedRuleNamespace,
+} from 'app/types/unified-alerting';
 import { PromAlertingRuleState } from 'app/types/unified-alerting-dto';
 
-import { isAlertingRule, isRecordingRule, isRecordingRulerRule, isGrafanaRulerRulePaused } from '../../utils/rules';
-
 interface Props {
-  includeTotal?: boolean;
-  group?: CombinedRuleGroup;
-  namespaces?: CombinedRuleNamespace[];
+  namespaces: CombinedRuleNamespace[];
 }
 
-const emptyStats = {
-  total: 0,
+// All available states for a rule need to be initialized to prevent NaN values when adding a number and undefined
+const emptyStats: Required<AlertGroupTotals> = {
   recording: 0,
-  [PromAlertingRuleState.Firing]: 0,
+  alerting: 0,
   [PromAlertingRuleState.Pending]: 0,
   [PromAlertingRuleState.Inactive]: 0,
   paused: 0,
   error: 0,
-} as const;
+  nodata: 0,
+};
 
-export const RuleStats = ({ group, namespaces, includeTotal }: Props) => {
-  const evaluationInterval = group?.interval;
-  const [calculated, setCalculated] = useState(emptyStats);
-
-  // Performance optimization allowing reducing number of stats calculation
-  // The problem occurs when we load many data sources.
-  // Then redux store gets updated multiple times in a pretty short period, triggering calculating stats many times.
-  // debounce allows to skip calculations which results would be abandoned in milliseconds
-  useDebounce(
-    () => {
-      const stats = { ...emptyStats };
-
-      const calcRule = (rule: CombinedRule) => {
-        if (rule.promRule && isAlertingRule(rule.promRule)) {
-          if (isGrafanaRulerRulePaused(rule)) {
-            stats.paused += 1;
-          }
-          stats[rule.promRule.state] += 1;
-        }
-        if (ruleHasError(rule)) {
-          stats.error += 1;
-        }
-        if (
-          (rule.promRule && isRecordingRule(rule.promRule)) ||
-          (rule.rulerRule && isRecordingRulerRule(rule.rulerRule))
-        ) {
-          stats.recording += 1;
-        }
-        stats.total += 1;
-      };
-
-      if (group) {
-        group.rules.forEach(calcRule);
-      }
+export const RuleStats = ({ namespaces }: Props) => {
+  const stats = { ...emptyStats };
 
-      if (namespaces) {
-        namespaces.forEach((namespace) => namespace.groups.forEach((group) => group.rules.forEach(calcRule)));
+  // sum all totals for all namespaces
+  namespaces.forEach(({ groups }) => {
+    groups.forEach((group) => {
+      const groupTotals = omitBy(group.totals, isUndefined);
+      for (let key in groupTotals) {
+        // @ts-ignore
```

Comment:
Can we figure this one out? No worries if not but I'd like to avoid `@ts-ignore` if we can